### DISCUSSION TITLE
Is there a good way to disable $PWD being set as the window title in the bash shell integration?

### DISCUSSION THREAD

**Discussion by tristan957** [ghostty-org/ghostty#2750] [Q&A]
Here are the relevant lines, I believe: https://github.com/ghostty-org/ghostty/blob/63bf16ff00ccac2f730519b4f0dd14cde857792d/src/shell-integration/bash/ghostty.bash#L144-L149.

cc @jparise if you have any thoughts on the matter. It would really be worthless to use PWD as the title when https://github.com/ghostty-org/ghostty/pull/2715 is merged.

**Comment by jparise** [comment]
Do you mean other than via `$GHOSTTY_SHELL_INTEGRATION_NO_TITLE = 1` and `shell-integration-features = no-title`?

**Reply by tristan957** [reply]
If you do that, you no longer get the process as the title of the window, eg `nvim`.

**Reply by mitchellh** [reply]
I haven’t looked at the bash integration. But it sounds like it may not be sending OSC7? The lines linked send OSC2 (set title) but what you really want is OSC7 (set pwd) unconditionally.

**Reply by jparise** [reply]
Yes, I agree with @mitchellh regarding unconditional OSC7.

But it also sounds like you'd need the current "title" behavior, which currently sets both command and title strings, to be split, allowing only the command and/or title to be set via OSC2?

**Reply by tristan957** [reply]
So do I need to send a PR which sends an unconditional OSC 7 and remove the OSC 2? Sorry, I'm not completely following.
